src/screens/ExchangeModal.js

```diff
@@ -517,6 +517,89 @@ export default function ExchangeModal({
     }
   };
 
+  const submit = useCallback(
+    async amountInUSD => {
+      setIsAuthorizing(true);
+      let NotificationManager = ios ? NativeModules.NotificationManager : null;
+      try {
+        const wallet = await loadWallet();
+        if (!wallet) {
+          setIsAuthorizing(false);
+          logger.sentry(`aborting ${type} due to missing wallet`);
+          return;
+        }
+
+        const callback = (success = false, errorMessage = null) => {
+          setIsAuthorizing(false);
+          if (success) {
+            setParams({ focused: false });
+            navigate(Routes.PROFILE_SCREEN);
+          } else if (errorMessage) {
+            Alert.alert(errorMessage);
+          }
+        };
+        logger.log('[exchange - handle submit] rap');
+        const nonce = await getNextNonce();
+        const swapParameters = {
+          chainId,
+          flashbots,
+          inputAmount,
+          nonce,
+          outputAmount,
+          tradeDetails,
+        };
+        const rapType = getSwapRapTypeByExchangeType(type);
+        await executeRap(wallet, rapType, swapParameters, callback);
+        logger.log('[exchange - handle submit] executed rap!');
+        analytics.track(`Completed ${type}`, {
+          aggregator: tradeDetails?.source || '',
+          amountInUSD,
+          inputTokenAddress: inputCurrency?.address || '',
+          inputTokenName: inputCurrency?.name || '',
+          inputTokenSymbol: inputCurrency?.symbol || '',
+          isHighPriceImpact: debouncedIsHighPriceImpact,
+          liquiditySources: tradeDetails?.protocols || [],
+          network: currentNetwork,
+          outputTokenAddress: outputCurrency?.address || '',
+          outputTokenName: outputCurrency?.name || '',
+          outputTokenSymbol: outputCurrency?.symbol || '',
+          priceImpact: priceImpactPercentDisplay,
+          slippage: slippageInBips / 100,
+          type,
+        });
+        // Tell iOS we finished running a rap (for tracking purposes)
+        NotificationManager &&
+          NotificationManager.postNotification('rapCompleted');
+      } catch (error) {
+        setIsAuthorizing(false);
+        logger.log('[exchange - handle submit] error submitting swap', error);
```

Comment:
here as well, can we set submitting state to false if something happens with loadWallet and it didn't load? i manually made it fail and the button is stuck in Submitting

Reply:
nvm